<?xml version="1.0" encoding="UTF-8"?> 
<launch>
  <!--
    main_rovr_teleop_direct.launch
    VERSION 0.0.0 Last Changed 2019-05-05
    Launch teleop joystick control over wifi, speaking directly to the motors,
    no improved control. 
  -->
  <!-- Interface: 
    Pub: 
      sent_messages (can_msgs/Frame)

    Sub:
      joy (sensor_msgs/Joy)
      recv_messages (can_msgs/Frame)
  -->

  <node pkg="joy" type="joy_node" name="joy_node" output="screen" respawn="true"/>

  <node pkg="bigrdigr_control" type="joy_to_onboard_direct_node" name="joy_to_onboard_direct_node" output="screen">
    <param name="frequency" value="2.0"/>
    <param name="linear_scale" value="1.0"/>
    <param name="angular_scale" value="1.0"/>
    <param name="lift_scale_up" value="0.25"/>
    <param name="lift_scale_down" value="0.05"/>
    <param name="trans_conv_scale" value="0.5"/>
    <param name="digger_scale" value="0.8"/>
    <param name="hold_conv_scale" value="1.0"/>
  </node>

  <node pkg="socketcan_bridge" type="socketcan_bridge_node" name="socketcan_bridge_node" output="screen"/>

</launch>
